Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/tan/Tan.kt

```diff
@@ -10,7 +10,7 @@ data class Tan(
 
     val areCharactersValid = allCharactersValid(value)
     val isTanValidFormat = value.length == MAX_LENGTH && isChecksumValid(value)
-    val isTanValid = areCharactersValid && isTanValidFormat
+    val isTanValid = if (value.length != MAX_LENGTH) true else areCharactersValid && isTanValidFormat
```

Comment:
please add some tests

Reply:
Also, shouldn't it rather be (value.length < MAX_LENGTH)

Reply:
just from a readabilty point of view id prefer if the check for no input is performed in a different place, isTanValid isn't reall true if there is no tan at all.